app/models/concerns/cms/addon/sns_poster.rb

```diff
@@ -0,0 +1,179 @@
+
+    def message_format(html)
+      html = ActionController::Base.helpers.strip_tags(html)
+      html = ActionController::Base.helpers.truncate(html, :length=> 253)
+    end
+
+    def use_twitter_post?
+      twitter_auto_post == "active"
+    end
+
+    def use_facebook_post?
+      facebook_auto_post == "active"
+    end
+
+    def twitter_url
+      "https://twitter.com/#{twitter_user_id}/status/#{twitter_post_id}" if use_twitter_post? && twitter_user_id.present? && twitter_post_id.present?
+    end
+
+    def facebook_url
+      "https://www.facebook.com/#{facebook_user_id}/posts/#{facebook_post_id}" if use_facebook_post? && facebook_user_id.present? && facebook_post_id.present?
+    end
+
+    def facebook_id_separator(facebook_param)
+      facebook_param.split("_")
+    end
+
+    def connect_twitter(snskeys)
+      Twitter::REST::Client.new do |config|
+#        config.consumer_key     = snskeys["consumer_key"]
+        config.consumer_key        = "#{@cur_site.twitter_consumer_key}"
+        config.consumer_secret     = "#{@cur_site.twitter_consumer_secret}"
+        config.access_token        = "#{@cur_site.twitter_access_token}"
+        config.access_token_secret = "#{@cur_site.twitter_access_token_secret}"
+      end
+    end
+
+    private
+      def post_sns
+        snskeys = SS.config.cms.sns_poster
+
+        # tweet
+        if use_twitter_post?
+          post_to_twitter(snskeys)
+        end
+
+        # facebook
+        if use_facebook_post?
+          post_to_facebook(snskeys)
+        end
+      end
+
+      def post_to_twitter(snskeys)
+        tweet = "#{name}｜#{full_url}"
+        client = connect_twitter(snskeys)
+        image_param = []
+        # 画像の添付があればuploadとupdateを用いて投稿
+        if file_ids.present?
+          i = 0
+          files.each do |file|
+            open(file.path, 'rb') do |f|
+              image_param << client.upload(f)
+            end
+            i += 1
+            break if i >= 4
+          end
+          twitter_param = client.update( tweet, { "media_ids"=> image_param.join(',') } )
+          # 画像の添付がなければupdateを用いて投稿
+        else
+          twitter_param = client.update(tweet)
+        end
+        # 戻り値から投稿IDを取得し、DBに保存
+        twitter_id = twitter_param.id
+        # URLを表示するためにスクリーンネームを取得し、DBに保存
+        user_screen_id = client.user.screen_name
+        self.set(twitter_post_id: twitter_id, twitter_user_id: user_screen_id)
+      end
+
+      def post_to_facebook(snskeys)
+        message = message_format(html)
+        if file_ids.present?
+          image_path = files.first.full_url
+        end
+#       graph = access_token_facebook(snskeys)
+        graph = Koala::Facebook::API.new(@cur_site.facebook_access_token)
+        # facebokに投稿し、戻り値を取得
+        facebook_params = graph.put_wall_post(
+          message, {
+          "name"=> "#{name} - #{site.name}",
+          "link"=> full_url,
+          "picture"=> image_path,
+          "description"=> description
+        })
```

Comment:
__[rubocop]__
* Indent the right brace the same as the start of the line where the left brace is.

<!-- Comment by [SideCI](https://sideci.com) -->